pkg/oc/clusterup/up.go

```diff
@@ -278,18 +279,45 @@ func (c *ClusterUpConfig) Complete(cmd *cobra.Command) error {
 		c.BaseDir = absHostDir
 	}
 
-	// Check if users are not trying to re-run cluster up on existing configuration
-	// but with different components enabled.
-	if !sets.NewString(c.UserEnabledComponents...).Equal(sets.NewString("*")) {
-		if _, err := os.Stat(filepath.Join(c.BaseDir, "etcd")); err == nil {
-			return fmt.Errorf("cannot use --enable when the cluster is already initialized, use cluster add instead")
+	if _, err := os.Stat(c.BaseDir); err != nil && os.IsNotExist(err) {
+		if err := os.MkdirAll(c.BaseDir, os.ModePerm); err != nil {
+			return fmt.Errorf("unable to create base directory %q: %v", c.BaseDir, err)
 		}
 	}
 
-	for _, currComponent := range knownComponents.UnsortedList() {
-		if isComponentEnabled(currComponent, componentsDisabledByDefault, c.UserEnabledComponents...) {
-			c.ComponentsToEnable = append(c.ComponentsToEnable, currComponent)
+	// When users run the cluster up for the first time we store the list of components the cluster up runs with
+	// into a "components.json" file inside base directory.
+	// On the second run, we won't allow users to specify new or delete existing components when the base directory
+	// was already initialized.
+	if _, err := os.Stat(filepath.Join(c.BaseDir, "components.json")); err == nil {
```

Comment:
@soltysh if the file exists, the `--enable` is ignored, but yes, telling user that it can't modify enabled components is better UIX

Reply:
Yup, not doing so would only confuse users, since they'd start complaining that they've specified some components with `--enable` and never seen them enabled on an already initialized cluster.